stb_image/Makefile

```diff
@@ -0,0 +1,46 @@
+# Port Metadata
+PORTNAME 	= stb_image
+PORTVERSION 	= 2.29
+
+MAINTAINER 	= Jason Rost (OniEnzeru)
+LICENSE 	= Public Domain
+SHORT_DESC 	= Single header library for image loading supporting many common formats.
+
+# No dependencies beyond the base system.
+DEPENDENCIES =
+
+# What files we need to download, and where from.
+DOWNLOAD_SITE 	= https://raw.githubusercontent.com/nothings/stb/master/
+# Format KOS-PORTS wants to work with
+DOWNLOAD_FILES 	= ${PORTNAME}-${PORTVERSION}.tar.gz
+# Actual downloaded file into dist ( we will package and tar this for compat )
+DOWNLOAD_FILESB = stb_image.h
+
+TARGET 		= libstb_image.a
+INSTALLED_HDRS 	= stb_image.h
+HDR_INSTDIR 	= stb_image
+
+# KOS Distributed extras (to be copied into the build tree)
+KOS_DISTFILES 	= KOSMakefile.mk stb_image.cpp
+
+include ${KOS_PORTS}/scripts/kos-ports.mk
+
+fetch:
+	@if [ ! -d "dist" ] ; then \
+		mkdir dist ; \
+	fi
+
+	@cd dist ; \
+	for _file in ${DOWNLOAD_FILESB}; do \
+		if [ ! -f "$$_file" ] ; then \
+			echo "fetchB"; \
+			echo "Fetching $$_file from ${DOWNLOAD_SITE} ..." ; \
+			file="${DOWNLOAD_SITE}/$$_file" ; \
+			${FETCH_CMD} $$file ; \
+		fi ; \
+		# Create proper directory structure for the .tar.gz and move header inside
+		mkdir -p ${PORTNAME}-${PORTVERSION};\
+		mv ${DOWNLOAD_FILESB} ${PORTNAME}-${PORTVERSION}/ ;\
+		# Tar the downloaded file so that KOS flags know what to do with the single header
+		tar -czvf ${PORTNAME}-${PORTVERSION}.tar.gz ${PORTNAME}-${PORTVERSION} ;\
+	done
```

Comment:
Just leaving a note for myself for future reference that I should probably find a way to clean this up... I know this is a bit of a strange example, but just in case we have other stb-type libraries that get added later, it might be a good idea to handle this kind of a case in the ports setup...